From 18a28d7f765a9729d51c83a379117add1cba17f1 Mon Sep 17 00:00:00 2001
From: "xinhui.pan" <xinhuix.pan@intel.com>
Date: Tue, 4 Aug 2015 10:54:24 +0800
Subject: [PATCH] cpufreq: introduce cpufreq_cpu_get_raw

Some cpufreq driver need policy through cpu. So we introduce it.
BUT it does not lock policy->rwsem itself, Pls. use it when you
are sure you has down_write/read the policy. Mostime, in
cpufreq driver callback, core has down_read policy->rwsem. This
function is copied from upstream.

Change-Id: I8ad322932d7f07769b9ff5e1a27d74a968643f54
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42456
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-2217
Signed-off-by: xinhui.pan <xinhuix.pan@intel.com>
---
 drivers/cpufreq/cpufreq.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index ab81519e5434..c4b2c6dd1a2b 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -191,6 +191,15 @@ unsigned int cpufreq_generic_get(unsigned int cpu)
 }
 EXPORT_SYMBOL_GPL(cpufreq_generic_get);
 
+/* Only for cpufreq core and some cpufreq callback with policy->rwsem locked*/
+struct cpufreq_policy *cpufreq_cpu_get_raw(unsigned int cpu)
+{
+	struct cpufreq_policy *policy = per_cpu(cpufreq_cpu_data, cpu);
+
+	return policy && cpumask_test_cpu(cpu, policy->cpus) ? policy : NULL;
+}
+EXPORT_SYMBOL_GPL(cpufreq_cpu_get_raw);
+
 struct cpufreq_policy *cpufreq_cpu_get(unsigned int cpu)
 {
 	struct cpufreq_policy *policy = NULL;
-- 
1.9.1

